src/main/java/frc/robot/autonomous/SimpleSwerveTrajectoryFollower.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.autonomous;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.trajectory.Trajectory;
+
+/** Implements a simple swerve trajectory follower, with a fixed robot heading */
+public class SimpleSwerveTrajectoryFollower extends TrajectoryFollower<ChassisSpeeds> {
+  private PIDController xController;
+  private PIDController yController;
+  private ProfiledPIDController angleController;
+
+  private Trajectory.State lastState = null;
+  private boolean finished = false;
+  /* Variable to track if calculateDriveSignal has run once yet */
+  protected boolean firstRun = false;
+
+  public SimpleSwerveTrajectoryFollower(
+      PIDController xController, PIDController yController, ProfiledPIDController angleController) {
+    this.xController = xController;
+    this.yController = yController;
+    this.angleController = angleController;
+    angleController.enableContinuousInput(0, 2 * Math.PI); // This works so far. Considering -pi..pi
+  }
+
+  @Override
+  protected ChassisSpeeds calculateDriveSignal(
+      Pose2d currentPose, Trajectory trajectory, double time, double dt) {
+    if (time > trajectory.getTotalTimeSeconds()) {
+      finished = true;
+      return new ChassisSpeeds();
+    }
+
+    // there is still time left!
+    lastState = trajectory.sample(time);
+    if (firstRun) {
+      angleController.reset(currentPose.getRotation().getRadians());
+      firstRun = false;
+    }
+    Pose2d poseRef = lastState.poseMeters;
+    double linearVelocityRefMeters = lastState.velocityMetersPerSecond;
+    double xFF = linearVelocityRefMeters * poseRef.getRotation().getCos();
+    double yFF = linearVelocityRefMeters * poseRef.getRotation().getSin();
+    // This sets the fixed angle goal to be 0. This would ideally be changed in a more advanced
+    // follower.
+    double angleFF = angleController.calculate(currentPose.getRotation().getRadians(), 0);
```

Comment:
This might make sense to integrate with the drive angle stuff, but that's not on dev - thoughts?

Reply:
Main use I see in trajectory is during a segment we may want to rotate 90 degrees over course of a path in which case we may want to provide an interpolable heading value for the duration of the path

otherwise I don't see why we wouldn't just used a fixed angle as changing the facing angle during the trajectory seems "sorta useless"

Reply:
I meant in the sense we shouldn't duplicate the pid controllers. What do profiled pid controllers offer?

Reply:
Ah, I see what you mean.
ProfiledPIDController allows us to constrain setpoints based on a trapezoidal motion profile for how we'd like our angle to change.
One could say that it's nice like this because it might help with autonomous trajectory consistency whereas in teleop intuitive mapping of controls->action is more highly prioritized.
But I do see your idea that these could be good to unify. In the version of `dev` this is built from, there was no existing angle PID controller yet, so that was the main reason it's a separate one.